### PR TITLE
 One-condition transcript quantification

### PR DESCRIPTION
**Pull request by yuukiiwa**
-Now, the pipeline has a transcript quantification process for cDNA and directRNA protocols, which can be skipped using the skip_transcriptquant flag. The transcriptquant flag can specify the method used for transcript quantification: bambu (default) or stringtie. 
-The information from ch_samplesheet_reformat is additionally channelled into ch_transquant_info, which is used to form ch_bambu_input (fasta and gtf) and ch_fasta_gtf (sample name, fasta, and gtf). 
-The SortBam process channels bam files into ch_sortbam_stringtie, which is joined to ch_fasta_gtf by sample name(s) to create the StringTie2 input ch_txome_reconstruction containing samples with bam files only.
-The SortBam process outputs bam files and bai files into directories bam and bam_index, respectively. The bam directory is channelled into ch_bamdir and ch_bamdir_fc, used in Bambu and FeatureCounts, respectively.

### CODE REVIEW

**Comment by yuukiiwa** [comment]
I downloaded [the test dataset for nanoseq](https://github.com/nf-core/test-datasets/tree/nanoseq) and modified the test.config and samplesheet_bc_dx.csv to test the bambu- and stringtie integrated pipeline. [The GTF file](https://hgdownload.soe.ucsc.edu/goldenPath/hg19/bigZips/genes/hg19.ensGene.gtf.gz) used is from UCSC.
test.config
Screen Shot 2020-07-15 at 12 17 51 PM
samplesheet_bc_dx.csv
Screen Shot 2020-07-15 at 12 18 23 PM
The Stringtie2 process completed successfully, but the pipeline stops at Stringtie2Merge as there were no transcript found in the Stringtie2 output:
stringtie
The bambu process stopped during bambu(). Here is the error message:
Screen Shot 2020-07-15 at 12 09 34 PM

**Comment by cying111** [comment]
Hi @yuukiiwa I have just created a new docker image for version 2 of bambu, could you try to see if it works? You can change the docker image to 0.2 version for bambu in the config file to use the recent docker image

**Comment by yuukiiwa** [comment]
Hi @cying111, I have changed the Bambu container to bambu:0.2, which allows the Bambu process to run. I am using SGNex_Hct116_directRNA_replicate6_run1.fastq.gz and SGNex_MCF7_directRNA_replicate1_run1.fastq.gz for testing the transcript quantification integration. The stringtie track completed successfully:
stringtie
Here is the error message from bambu:
Screen Shot 2020-07-20 at 3 36 59 PM

**Comment by yuukiiwa** [comment]
I have added in a bam file input option for running nanoseq and tested it with the stringtie track:
fastq input:
fastq_stringtie
bam input:
bam_stringtie

File: docs/usage.md [hunk 1]
```diff
@@ -291,6 +291,7 @@ nextflow run nf-core/nanoseq \
     --protocol cDNA \
     --skip_basecalling \
     --skip_demultiplexing \
+    --skip_qc \
```
Comment (cying111) [inline docs/usage.md:294]: maybe --skip_alignment is better?

Comment (yuukiiwa) [reply docs/usage.md:294]: I added both skip_qc (line 294) and skip_alignment (line 295) because the qc steps are for fastq files.

**Pull request merged**